recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -97,215 +113,223 @@ class GdalConan(ConanFile):
+        copy(self, "*.cmake",
+             src=os.path.join(self.recipe_folder, "cmake"),
+             dst=os.path.join(self.export_sources_folder, "src", "cmake", "helpers"))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if is_apple_os(self):
+            # FIXME: compilation fails on macOS due to difference in ICONV_CONST value in iconv() signature
+            # https://github.com/conan-io/conan-center-index/pull/19298#issuecomment-1840433415
+            # https://github.com/roboticslibrary/libiconv/blob/v1.17/include/iconv.h.in#L82
+            # https://lists.gnu.org/archive/html/bug-gnu-libiconv/2008-07/msg00005.html
+            self.options.with_libiconv = False
+        if Version(self.version) < "3.7":
+            # Latest versions of Arrow are no longer compatible with GDAL 3.5
+            self.options.with_arrow = False
+        if Version(self.version) < "3.8":
+            del self.options.with_libaec
 
     def configure(self):
-        if self.options.with_crypto != "deprecated":
-            self.output.error("with_crypto option is deprecated, use with_openssl instead.")
-
         if self.options.shared:
-            try:
-                del self.options.fPIC
-            except:
-                pass
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("json-c/0.16")
+        self.requires("json-c/0.17")
         self.requires("libgeotiff/1.7.1")
-
+        self.requires("libtiff/4.6.0")
+        self.requires("proj/9.3.1")
+        # Used in a public header here:
+        # https://github.com/OSGeo/gdal/blob/v3.7.1/port/cpl_minizip_ioapi.h#L26
+        self.requires("zlib/[>=1.2.11 <2]", transitive_headers=True, transitive_libs=True)
         if self.options.with_armadillo:
-            self.requires("armadillo/10.7.3")
-
+            self.requires("armadillo/12.6.4")
         if self.options.with_arrow:
-            self.requires("arrow/8.0.1")
-
+            self.requires("arrow/14.0.2")
+        if self.options.with_basisu:
+            self.requires("libbasisu/1.15.0")
         if self.options.with_blosc:
-            self.requires("c-blosc/1.21.1")
-
+            self.requires("c-blosc/1.21.5")
+        if self.options.with_brunsli:
+            self.requires("brunsli/cci.20231024")
         if self.options.with_cfitsio:
-            self.requires("cfitsio/4.1.0")
-
+            self.requires("cfitsio/4.3.0")
         if self.options.with_cryptopp:
-            self.requires("cryptopp/8.7.0")
-
+            self.requires("cryptopp/8.8.0")
         if self.options.with_curl:
-            self.requires("libcurl/8.2.0")
-
+            self.requires("libcurl/[>=7.78 <9]")
         if self.options.with_dds:
             self.requires("crunch/cci.20190615")
-
+        if self.options.with_ecw:
+            self.requires("libecwj2/3.3")
```

Comment:
This requirement does not exist in CCI:
```
$ conan install --requires=gdal/3.8.2 -b=gdal/3.8.2 -b=missing -r=pr-19298 -r=conancenter -o "&:with_ecw=True"

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
[options]
&:with_ecw=True

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos


======== Computing dependency graph ========
libecwj2/3.3: Not found in local cache, looking in remotes...
libecwj2/3.3: Checking remote: pr-19298
libecwj2/3.3: Checking remote: conancenter
Graph root
    cli
Requirements
    arrow/14.0.2#cbdce5920d1bbec1dc4782780d71a8bd - Cache
    gdal/3.8.2#fafaf4aff0208d8656482b36f0a9a88c - Cache
    jbig/20160605#2d29fa02aacd76902e0d2cbbc24631ef - Cache
    json-c/0.17#f863bad3e5c80dc90679092f67bbd680 - Cache
    libcurl/8.5.0#4fc73f2d68bae7c25d11a8b21d52233d - Cache
    libdeflate/1.19#3ea74a4549efc14d4b1202dc4bfbf602 - Cache
    libgeotiff/1.7.1#ef2a55dbc032ef6d13598247e2e8fe4b - Cache
    libjpeg/9e#68269859e4325ddc3f995d1fd3fc9187 - Cache
    libtiff/4.6.0#44d09b1f75a7fd97d4f7daea5e7aed8e - Cache
    libwebp/1.3.2#52f69c4a31c5cf033fdd9230d77a8e38 - Cache
    nlohmann_json/3.11.3#45828be26eb619a2e04ca517bb7b828d - Cache
    proj/9.3.1#da0fce97195990d35c992a45aeabe766 - Cache
    sqlite3/3.44.2#6aea4802c643c19f97b7f5a8a4b4f16d - Cache
    xsimd/9.0.1#f93b209d2cb92ac4b73e994e65c66aa1 - Cache
    xz_utils/5.4.5#a26a2d42ab9c3dabc151bbe1bf898e53 - Cache
    zlib/1.3#06023034579559bb64357db3a53f88a4 - Cache
    zstd/1.5.5#b87dc3b185caa4b122979ac4ae8ef7e8 - Cache
Build requirements
    autoconf/2.71#53be95d228b2dcb30dc199cb84262d8f - Cache
    automake/1.16.5#058bda3e21c36c9aa8425daf3c1faf50 - Cache
    cmake/3.28.1#92f79424d7b65b12a84a2180866c3a78 - Cache
    gnu-config/cci.20210814#dc430d754f465e8c74463019672fb97b - Cache
    libtool/2.4.7#525321775eb6780866f0bd534424aea1 - Cache
    m4/1.4.19#b38ced39a01e31fef5435bc634461fd2 - Cache
    meson/1.2.2#aace9dcc1db58fa42ecb5292f724092d - Cache
    ninja/1.11.1#77587f8c8318662ac8e5a7867eb4be21 - Cache
    pkgconf/2.1.0#27f44583701117b571307cf5b5fe5605 - Cache
    sqlite3/3.44.2#6aea4802c643c19f97b7f5a8a4b4f16d - Cache
Resolved version ranges
    cmake/[>=3.16 <4]: cmake/3.28.1
    cmake/[>=3.18 <4]: cmake/3.28.1
    libcurl/[>=7.78.0 <9]: libcurl/8.5.0
    zlib/[>=1.2.11 <2]: zlib/1.3
ERROR: Package 'libecwj2/3.3' not resolved: Unable to find 'libecwj2/3.3' in remotes.
```

Reply:
Currently, but I opened a PR for it: #21341.

Reply:
Ohh right, thanks a lot, missed it :)

Reply:
While that one get merged, I think it will be faster to have this one commented/validated out and a new PR can be made once the dependency is ready